agw/main.tf

```diff
@@ -0,0 +1,78 @@
+locals {
+  resource_group = "${var.environment}_${var.application_name}_gateway"
+}
+
+resource "azurerm_resource_group" "rg" {
+  name     = "${local.resource_group}"
+  location = "${var.location}"
+}
+
+resource "azurerm_subnet" "subnet" {
+  name                 = "${var.environment}_${var.application_name}_agw_subnet"
+  resource_group_name  = "${var.vnet_rg_name}"
+  virtual_network_name = "${var.vnet_name}"
+  address_prefix       = "${var.agw_subnet}"
+}
+
+resource "azurerm_public_ip" "agw_pip" {
+  name                         = "${var.environment}_${var.application_name}_agw_pip"
+  location                     = "${var.location}"
+  resource_group_name          = "${local.resource_group}"
+  public_ip_address_allocation = "dynamic"
+  depends_on                   = ["azurerm_resource_group.rg"]
+}
+
+resource "azurerm_application_gateway" "network" {
+  name                = "${var.environment}_${var.application_name}-gateway-12345"
+  resource_group_name = "${azurerm_resource_group.rg.name}"
+  location            = "${var.location}"
+
+  sku {
+    name     = "${var.agw_size}"
+    tier     = "${var.tier}"
+    capacity = "${var.capacity}"
+  }
+
+  gateway_ip_configuration {
+    name      = "my-gateway-ip-configuration"
+    subnet_id = "${azurerm_subnet.subnet.id}"
+  }
+
+  frontend_port {
```

Comment:
Does the AppGateway accept only one frontend port per AppGW resource? Can this be changed to accept multiple front end ports if possible?

Reply:
We had this discussion earlier in the day.  It merits more discussion, but it can handle more than one port, but you have to define that in code.  In other words there is no way to loop of that data today.  We could potentially use an ARM template, but it will take several hours to work out that code.  Let me know how you'd like to proceed.